graphql-compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/GraphqlFunctionValidator.java

```diff
@@ -173,14 +174,22 @@ private void validateInputParamType(FunctionDefinitionNode functionDefinitionNod
     }
 
     private boolean hasInvalidInputParamType(ParameterSymbol inputTypeSymbol) {
-        return !hasPrimitiveType(inputTypeSymbol.typeDescriptor());
+        boolean hasInvalidParamType = true;
```

Comment:
This variable definition can be removed if we directly return `false` when we find an invalid input type.